Documentation/Rust Style.md

```diff
@@ -36,6 +36,29 @@ Apply `rustfmt` to new code before committing, using the default configuration o
 
 In `impl`s, always put public functions before private ones.
 
+## Bringing names into scope (`use` statements)
+
+Generally `use` statements should be employed to bring names from different modules into scope. However, functions from other modules should not be brought fully into scope. Instead their module should be brought into scope meaning that subsequent usage of the function requires one level of qualification. For example, if we have:
```

Comment:
I wonder what's the motivation though? :thinking: 
I guess probably it's readability or easier to avoid name clash or both, etc. But could we make it explicit? :) 
Seems like a couple of explicit examples would make it easier to understand why we're doing this.

Reply:
I think the motivation is that many functions can be disambiguated pretty easily this way, particularly given that we're discouraged from "stuttering" when naming them.  For example, we could have `encode()` as a function in modules `hex`, `base32` and `base64`.  That function shouldn't be named e.g. `hex::hex_encode()`, so when we use it, it's clearer to write `hex::encode()` rather than just `encode()`.